api/src/user/user.handlers.ts

```diff
@@ -226,8 +227,84 @@ export const getUserWorksheet = async (
 
   const allWorksheets = worksheetListToMap(userWorksheets);
 
+  // Collect all unique CRNs grouped by season
+  const seasonCrnMap = new Map<string, Set<number>>();
+  for (const worksheetsBySeason of Object.values(allWorksheets)) {
+    for (const [seasonCode, worksheetsByNum] of Object.entries(
+      worksheetsBySeason,
+    )) {
+      if (!seasonCrnMap.has(seasonCode))
+        seasonCrnMap.set(seasonCode, new Set());
+
+      for (const worksheet of Object.values(worksheetsByNum)) {
+        for (const course of worksheet.courses)
+          seasonCrnMap.get(seasonCode)!.add(course.crn);
+      }
+    }
+  }
+
+  // Fetch same_course_id for all CRNs using GraphQL
+  const crnToSameCourseId = new Map<string, number>();
+  for (const [seasonCode, crns] of seasonCrnMap.entries()) {
+    if (crns.size === 0) continue;
+
+    const query = gql`
+      query GetSameCourseIds($crns: [Int!]!, $season: String!) {
+        listings(
+          where: { crn: { _in: $crns }, season_code: { _eq: $season } }
+        ) {
+          crn
+          course {
+            same_course_id
+          }
+        }
+      }
+    `;
```

Comment:
I believe we don't do this and you are supposed to add this to a `.queries.graphql` file, so graphql-codegen can automatically bootstrap the SDK. See `challenge.queries.graphql` and its generated `challenge.queries.ts` for example